std/algorithm/iteration.d

```diff
@@ -5185,3 +5185,100 @@ if (isRandomAccessRange!Range && hasLength!Range)
          [1, 2, 0],
          [2, 1, 0]]));
 }
+/*
+Returns the array without any duplicates, keeping the initial order of appearance.
+If $(D T) is a basic type, a struct or a class which overrides the $(D toHash) operator
+the resulting array is computed in $(D O(n)) steps. If $(D T) is a class which does not
+override the $(D toHash) operator, the result is computed in $(D O(n^2)) steps. Aditionally,
+if the $(D opEquals) operator is not overriden for classes, $(D noDupes) considers the class
+pointer for duplication removal.
+*/
+T[] noDupes(T)(T[] s)
+{
+    static if (is(T == class) && !__traits(isOverrideFunction, T.toHash))
+    {
+        import std.algorithm : canFind;
+        T[] result;
+        foreach (T elem; s)
```

Comment:
first element of `s` can be inserted, always, unconditionally. (in the other else part of the static branch too)

Reply:
this would imply checking manually for an empty array, instead the foreach does that for you, plus the code is more readable this way